### PR TITLE
Replace hardcoded slashes with GetPathDelimiter()

### PR DESCRIPTION
**Pull request by anosach-luxoft**

### CODE REVIEW

**Comment by AGaliuzov** [comment]
@anosach-luxoft , looks not too bad, but maybe add some method called `combine_path` or any
and hide this `GetDelimter` function within this new method?

**Comment by malirod** [comment]
agree with @AGaliuzov.

**Comment by AGaliuzov** [comment]
Reviewed and tested seems workable

**Comment by anosach-luxoft** [comment]
@malirod, @AGaliuzov, @pestOO review has been updated

File: src/components/utils/src/file_system_posix.cc [hunk 1]
```diff
@@ -462,11 +462,21 @@ void file_system::MakeAbsolutePath(std::string& path) {
...
 }
 
+std::string file_system::ConcatPath(const std::string& str1,
+                                    const std::string& str2) {
+  return str1 + GetPathDelimiter() + str2;
+}
+std::string file_system::ConcatPath(const std::string& str1,
+                                    const std::string& str2,
+                                    const std::string& str3) {
+  return str1 + GetPathDelimiter() + str2 + GetPathDelimiter() + str3;
```
Comment (pestOO) [inline src/components/utils/src/file_system_posix.cc]: May be `return ConcatPath(ConcatPath(str1, str2), str3))`?

File: src/components/utils/src/file_system_win.cc [hunk 2]
```diff
@@ -430,4 +430,14 @@ std::string file_system::GetPathDelimiter() {
...
 
+std::string file_system::ConcatPath(const std::string& str1,
+                                    const std::string& str2) {
+  return str1 + GetPathDelimiter() + str2;
+}
+std::string file_system::ConcatPath(const std::string& str1,
+                                    const std::string& str2,
+                                    const std::string& str3) {
+  return str1 + GetPathDelimiter() + str2 + GetPathDelimiter() + str3;
+}
```
Comment (pestOO) [inline src/components/utils/src/file_system_win.cc:441]: It's not cool to duplicate this.
Possible need to extract support and platform-specific classes.

**Comment by pestOO** [comment]
Reviewed

**Comment by anosach-luxoft** [comment]
@malirod, we cannot use C++11 standart. Please do not propose C++11 features

File: src/components/utils/src/file_system_posix.cc (hunk 1 above)
Comment (anosach-luxoft) [inline src/components/utils/src/file_system_posix.cc]: @pestOO, the same as above

File: src/components/utils/src/file_system_win.cc (hunk 2 above)
Comment (anosach-luxoft) [inline src/components/utils/src/file_system_win.cc:441]: @pestOO, agree with you, but let's do it not within this task

**Comment by anosach-luxoft** [comment]
@AGaliuzov, see baf1bda. And I don't like to use `inline` directive. Let's keep this thing up to compiler.

**Comment by AGaliuzov** [comment]
@anosach-luxoft , any way it is up to compiler to create inline function.

**Pull request merged**